packages/cli/src/runners/task-managers/data-request-response-stripper.ts

```diff
@@ -81,11 +82,31 @@ export class DataRequestResponseStripper {
 	}
 
 	private stripInputData(inputData: ITaskDataConnections): ITaskDataConnections {
-		if (this.stripParams.input) {
+		if (!this.stripParams.input.include) {
+			return {};
+		}
+
+		return this.stripParams.input.chunk ? this.stripChunkedInputData(inputData) : inputData;
+	}
+
+	private stripChunkedInputData(inputData: ITaskDataConnections): ITaskDataConnections {
+		const { chunk } = this.stripParams.input;
+		a.ok(chunk);
+
+		const inputItems = inputData.main?.[0];
+		if (!inputItems) {
 			return inputData;
 		}
 
-		return {};
+		// If a chunk of the input data is requested, we only return that chunk
+		// It is the responsibility of the requested to rebuild the input data
+		const chunkInputItems = inputItems.slice(chunk.startIndex, chunk.startIndex + chunk.count);
+		const chunkedInputData: ITaskDataConnections = {
+			...inputData,
```

Comment:
I'm generally not a big fan of mutation as it can lead to unexpected behaviour and cause weird and hard to debug issues, unless there's a clear reason (e.g. performance or internal data). Here the data is coming from outside and I don't see this as being a performance issue, so I'll keep it like this